backend/oidc/models.py

```diff
@@ -43,3 +41,27 @@ def is_active_refresh_token(self):
         ):
             return False
         return True
+
+
+class OIDCProfile(AbstractOIDCProfile):
+    user = models.OneToOneField(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        verbose_name=_("user"),
+        related_name="oidc_profile",
+    )
+
+
+class EAuthorizationProfile(AbstractOIDCProfile):
+    oidc_profile = models.OneToOneField(
+        OIDCProfile,
+        on_delete=models.CASCADE,
+        verbose_name=_("oidc_profile"),
+        related_name="eauthorization_profile",
+    )
+    company = models.OneToOneField(
```

Comment:
Would EAuthorizationProfile be reusable for helsinki-benefit too? For example, I remember discussing with @quyenlq that he had plans to reuse at least some of the code for Company model, but e.g. if there's an abstract Company base then we can't have foreign key to that?

Reply:
True, it'd be great if Helsinki-benefit can reuse part of the Company and this model too.
@klemmari1  Would it be possible for you to make the Company model abstract and inherit it later in Kesaseteli app with its own model. We can discuss later which fields should be included in the abstract Company class later

Reply:
True that. I did not think it quite through but yes it should be abstracted.